app/notifications/rest.py

```diff
@@ -32,6 +35,150 @@
+                "SES callback failed: message missing"
+            )
+            return jsonify(
+                result="error", message="SES callback failed: message missing"
+            ), 400
+
+        if 'notificationType' not in ses_request['Message']:
+            current_app.logger.error(
+                "SES callback failed: notificationType missing"
+            )
+            return jsonify(
+                result="error", message="SES callback failed: notificationType missing"
+            ), 400
+
+        status = ses_response_status.get(ses_request['Message']['notificationType'], None)
+        if not status:
+            current_app.logger.info(
+                "SES callback failed: status {} not found.".format(status)
+            )
+            return jsonify(
+                result="error",
+                message="SES callback failed: status {} not found".format(ses_request['Message']['notificationType'])
+            ), 400
+
+        try:
+            source = ses_request['Message']['mail']['source']
+            if is_not_a_notification(ses_request['Message']['mail']['source']):
+                current_app.logger.info(
+                    "SES callback for notify success:. source {} status {}".format(source, status['notify_status'])
+                )
+                return jsonify(
+                    result="success", message="SES callback succeeded"
+                ), 200
+
+            reference = ses_request['Message']['mail']['messageId']
+            if notifications_dao.update_notification_status_by_reference(reference, status['notify_status']) == 0:
+                current_app.logger.info(
+                    "SES callback failed: notification not found. Status {}".format(status['notify_status'])
+                )
+                return jsonify(
+                    result="error",
+                    message="SES callback failed: notification not found. Status {}".format(status['notify_status'])
+                ), 404
+            return jsonify(
+                result="success", message="SES callback succeeded"
+            ), 200
+
+        except KeyError:
+            current_app.logger.error(
+                "SES callback failed: messageId missing"
+            )
+            return jsonify(
+                result="error", message="SES callback failed: messageId missing"
+            ), 400
+
+    except ValueError as ex:
+        current_app.logger.exception(
+            "SES callback failed: invalid json {}".format(ex)
+        )
+        return jsonify(
+            result="error", message="SES callback failed: invalid json"
+        ), 400
+
+
+def is_not_a_notification(source):
+    invite_email = "{}@{}".format(
+        current_app.config['INVITATION_EMAIL_FROM'],
+        current_app.config['NOTIFY_EMAIL_DOMAIN']
+    )
+    if current_app.config['VERIFY_CODE_FROM_EMAIL_ADDRESS'] == source:
+        return True
+    if invite_email == source:
```

Comment:
It just looked really evil like that, thought might be cleaner? Happy to change if not.